app/javascript/src/components/Forecast/Forecast.tsx

```diff
@@ -1,13 +1,21 @@
-import React, { useState } from 'react';
-import { any } from 'prop-types';
-import { ApolloProvider } from 'react-apollo';
-import ApolloClient from '../../api/ApolloClient';
+import React, { FC, useState } from 'react';
+import { ApolloProvider } from '@apollo/client';
 
-import PointOfInterestList from './PointOfInterestList';
-import ForecastMap from './ForecastMap';
-import ForecastSelector from './ForecastSelector';
+import { PointOfInterestList } from './PointOfInterestList';
+import { ForecastMap } from './ForecastMap';
+import { ForecastSelector } from './ForecastSelector';
 
-export default function Forecast({ options: { googleApiKey } }) {
+import ApolloClient from '../../api/ApolloClient'
+
+interface ForecastProps {
+  options: Key,
+}
+
+interface Key {
+  googleApiKey: string,
+}
+
+export const Forecast: FC<ForecastProps> = ({ options: { googleApiKey } }) => {
```

Comment:
not knowing how typescript works, could we not have an interface that had both `options: { googleApiKey}` ? or are nested things not a thing? or not preferred ? also on the destructuring side we have `{ options: { googleApiKey } }` can this not reuse some kind of type? again not knowing myself :)

Reply:
we can do this as well. If options will only have the googleApiKey, I think this might be a better alternative, if there were more attributes, I'd prefer having a separate interface, but again, just personal preference

```
interface ForecastProps {
  options: {
    googleApiKey: string,
  },
}
```

_also on the destructuring side we have { options: { googleApiKey } } can this not reuse some kind of type?_ not see I understand what you mean